chsdi/templates/htmlpopup/oev_haltestellen.mako

```diff
@@ -0,0 +1,151 @@
+<%inherit file="base.mako"/>
+
+<%def name="table_body(c, lang)">
+    <%!
+        from pyramid.url import route_url
+        import chsdi.lib.helpers as h
+    %>
+    <%
+        protocol = request.scheme
+        lang = request.lang
+        topic = request.matchdict.get('map')
+        host = h.make_agnostic(request.host_url + request.uscript_name)
+
+        id = c['attributes']['nummer']
+        c['baseUrl'] = h.make_agnostic(''.join((protocol, '://', request.registry.settings['geoadminhost'])))
+    %>
+<style>
+  tr:last-child {
+    display:none;
+  }
+</style>
+<script>
+
+var timesRun = 0;
+var dest;
+function getInfos(val){
+    if (val) {
+      dest = val;
```

Comment:
Of course, the globals has to go.

Not sure if to put it into geoadmin is good as it creates stong coupling. the htmlpopup call might be used and inserted elsewhere (other sites) as well and in case of the extendedhtmlpopup needs to be standalone anyway.

That being said, I share your concern about testability; but there might be other ways to address that. I envision more feature rich popups in the future, and we need to find a good solution.